.github/workflows/main.yml

```diff
@@ -68,25 +68,84 @@ jobs:
   website:
     name: build website
     runs-on: ubuntu-latest
+    defaults:
+      # use login shell to make use of .bash_profile /.bashrc
+      run:
+        shell: bash -l {0}
     steps:
+    - name: Cancel Previous Runs
+      uses: styfle/cancel-workflow-action@0.6.0
+      with:
+        access_token: ${{ github.token }}
     - uses: actions/checkout@v2
       with:
         fetch-depth: 0
         submodules: "recursive"
-    - name: Set up Python
-      uses: actions/setup-python@v2
+    - name: Cache VisPy Demo and Test Data
+      uses: actions/cache@v2
+      env:
+        cache-name: cache-vispy-data-website
+        test-version: 1
```

Comment:
I think hard coding values is fine for now to get things to work. But really 19 MB is trivial to download each time, so maybe we should just scrap it?

Reply:
> Any thoughts on caching the gallery directory generated by SG?

I wouldn't cache this. A sphinx build of 8 minutes (including setup!) is very fast. While Sphinx and SG try to be smart about what to rebuild it doesn't seem worth the hassle here

Reply:
True. We may want to consider it in the future. Right now this isn't rendering a majority of vispy's examples.

Reply:
If speed is eventually a concern, rather than caching I'd go for what MNE does and just build examples that are changed in a given PR (it's very easy to make a trivial change to trigger this if you want to see one) and on `main` always build all from scratch (`main` commits are rare enough that it's not too bad).

Reply:
The issue with not testing all examples on PRs is that vispy is lacking a lot of basic unit tests and the examples are the only places some pieces of code are used and tested. Also, there is a chance that someone changing a Visual interface wouldn't think to check for examples (or might miss one since there are so many) so we wouldn't detect a broken example until it was merged to main.

Reply:
In that case I wouldn't cache the generated directery *or* do anything special to limit what's executed, I'd just let them all build completely every time.

If there are a few examples that take a long time (> 30 s?) to build, you can add something like `_slow` to the filename for example, and not build just this select few on PRs (by using a negative lookahead filename_pattern) but always build everything on main commits (or on PRs where people put `[circle full]` in the commit title for example).

Reply:
Yeah, I suppose there is endless fanciness we could get into with this. Maybe I'll just leave it for now. I'll wait for others to review. I'm ok leaving the limited caching that I have in here now. It's something. Oh but I have some commented out code in there I'll remove that.